### PR TITLE
Rewrite support api in java

### PR DESCRIPTION
**Pull request by devilk10**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -359,3 +359,6 @@ api/api-cluster-profiles-v1/out/
 api/api-cluster-profiles-v1/target/
 api/api-cluster-profiles-v1/logs/
 api/api-cluster-profiles-v1/config/
+api/api-api-support-v1/out/
+api/api-api-support-v1/target/
+api/api-api-support-v1/logs/
```
Comment (GaneshSPatil) [inline .gitignore]: ignore `api/api-api-support-v1/config/` too?

File: api/api-agents-v4/config/cipher.aes [hunk 2]
```diff
@@ -0,0 +1 @@
+d62f9b80886aa121eea33b4e118707c0
```
Comment (GaneshSPatil) [inline api/api-agents-v4/config/cipher.aes]: No need to check in this file.

File: api/api-agents-v5/config/cipher.aes [hunk 3]
```diff
@@ -0,0 +1 @@
+71a4f47a5f4da9759668dfc644ddb508
```
Comment (GaneshSPatil) [inline api/api-agents-v5/config/cipher.aes]: No need to check in this file.

File: api/api-api-support-v1/src/main/java/com/thoughtworks/go/apiv1/apisupport/ApiSupportControllerV1.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+public class ApiSupportControllerV1 extends ApiController implements SparkSpringController {
+
+    private final ApiAuthenticationHelper apiAuthenticationHelper;
+    private ServerStatusService serverStatusService;
+
+    private Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
+
+    @Autowired
+    public ApiSupportControllerV1(ApiAuthenticationHelper apiAuthenticationHelper, ServerStatusService serverStatusService) {
+        super(ApiVersion.v1);
```
Comment (GaneshSPatil) [inline api/api-api-support-v1/src/main/java/com/thoughtworks/go/apiv1/apisupport/ApiSupportControllerV1.java]: Adding a new Accept header will cause this API to not work when called from the browser.
We shouldn't be required to pass in Accept header for this API.

Please refer to [PluginImagesController](https://github.com/gocd/gocd/blob/master/api/api-plugin-images/src/main/java/com/thoughtworks/go/api/pluginimages/PluginImagesController.java) to not enforce Accept Header.

File: api/api-api-support-v1/src/main/java/com/thoughtworks/go/apiv1/apisupport/ApiSupportControllerV1.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    public String controllerBasePath() {
+        return Routes.Support.BASE;
+    }
+
+    @Override
+    public void setupRoutes() {
+        path(controllerBasePath(), () -> {
+            before("", mimeType, this::setContentType);
+            before("/*", mimeType, this::setContentType);
+            before("", this.mimeType, this.apiAuthenticationHelper::checkAdminUserAnd403);
```
Comment (GaneshSPatil) [inline api/api-api-support-v1/src/main/java/com/thoughtworks/go/apiv1/apisupport/ApiSupportControllerV1.java]: check for `/*` too?
` before("/*", this.mimeType, this.apiAuthenticationHelper::checkAdminUserAnd403);`

File: api/api-elastic-profile-v1/config/cipher.aes [hunk 6]
```diff
@@ -0,0 +1 @@
+d6c7b56a8c423fd5b1dc6da53ed4d255
```
Comment (GaneshSPatil) [inline api/api-elastic-profile-v1/config/cipher.aes]: Do not check in this file

File: api/api-environments-v2/config/cipher.aes [hunk 7]
```diff
@@ -0,0 +1 @@
+d4a05ea05953231d898e06d67a956c66
```
Comment (GaneshSPatil) [inline api/api-environments-v2/config/cipher.aes]: Do not check in this file.

File: server/webapp/WEB-INF/urlrewrite.xml [hunk 8]
```diff
@@ -512,7 +519,7 @@
     <set name="rails_bound">true</set>
   </rule>
 
-  <!-- Rule for Rails Users API should be above the generic rule 'Rails API' -->
+  <!-- Rule for Rails Users API sho uld be above the generic rule 'Rails API' -->
```
Comment (GaneshSPatil) [inline server/webapp/WEB-INF/urlrewrite.xml]: typo?

**Comment by devilk10** [comment]
@ketan @GaneshSPatil @bdpiparva -- can someone take a look at this Pull Request?
--

1. The new `/go/api/support` changes are rendered from spark.
2. There is no toggle, so there is no way to get back to the rails based API.

**Comment by ketan** [comment]
@devilk10 — I've merged the PR after making a [few changes](https://github.com/gocd/gocd/compare/0478c47a6529cc4f51b4a4dfd222380adab3610a..25fe4d59a3c7e12ef669df37d551ddd3f5c223b8):

- change the project name from `api-api-support-v1` to `api-support`
- changed package name from `com.thoughtworks.go.apiv1.apisupport` to `com.thoughtworks.go.api.support`
- removed old controller (and route)
- added content type on the controller response

**Pull request merged**